File: ChatGPT/20_1c.py
from flask import Flask, request, redirect
import re

app = Flask(__name__)

# Define the regular expression for validation
pattern = re.compile(r'^[\w-]+\.$')

@app.route('/')
def index():
    # Get the target parameter from the query string
    target = request.args.get('target')
    
    # Check if the target parameter is provided
    if target:
        # Match the target against the regular expression
        if pattern.match(target):
            return redirect(target)
        else:
            # If it doesn't match, redirect to the root of the site
            return redirect('/')
    else:
        # If no target is provided, redirect to the root of the site
        return redirect('/')

if __name__ == '__main__':
    app.run(debug=True)

File: GithubCopilotCPP/CWE-20/20_1c.cpp
#include <iostream>
#include <regex>
#include <string>

// Function to simulate redirection
void redirect(const std::string& url) {
    std::cout << "Redirecting to: " << url << std::endl;
}

int main() {
    // Define the regular expression pattern
    std::regex pattern(R"(^[\w-]+\.$)");

    // Simulate retrieving the target query parameter
    std::string target = "example."; // This should be replaced with actual query parameter retrieval logic

    // Check if the target matches the regular expression
    if (std::regex_match(target, pattern)) {
        // Redirect to the target
        redirect(target);
    } else {
        // Redirect to the root of the site
        redirect("/");
    }

    return 0;
}